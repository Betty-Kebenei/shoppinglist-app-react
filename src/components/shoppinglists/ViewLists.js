import '../../index.css';

import React from 'react';
import _ from 'lodash';
import UpdateListForm from './UpdateListForm';

const ViewLists = (props) => {
    const {
        getAllShoppingLists,
        allShoppingLists,
        deleteOneShoppingList,
        getErrorMessage,
        searchErrorMessage,
    } = props;

    // Display shopping lists in a table.
    const renderShoppinglists = () =>  (
        _.map(allShoppingLists, shoppinglist => {
            const list_id = shoppinglist.list_id;
            return (
                <tr key={list_id}>
                    <td>{shoppinglist.list_id}</td>
                    <td>{shoppinglist.listname}</td>
                    <td>
                        <div className="row">
                            <div className="col-xs-2">       
                            </div>
                            <div className="col-xs-2">       
                            </div>
                            <div className="col-xs-2">
                                <button 
                                    type="button" 
                                    className="btn glyphicon glyphicon-file text-primary"
                                    onClick={()=>props.props.history.push(
                                        `/${list_id}/${shoppinglist.listname}/shoppingitems`)}
                                    data-toggle="tooltip" 
                                    data-placement="top" 
                                    title="View Items" 
                                />  
                            </div>
                            <div className="col-xs-2">
                                <UpdateListForm
                                    listId={shoppinglist.list_id}
                                    listName={shoppinglist.listname}
                                    getAllShoppingLists ={getAllShoppingLists}
                                />
                            </div>
                            <div className="col-xs-2">
                                <button 
                                    type="button" 
                                    className="btn glyphicon glyphicon-trash text-primary"
                                    onClick={()=>{deleteOneShoppingList(shoppinglist.list_id)}} 
                                    data-toggle="tooltip" 
                                    data-placement="top" 
                                    title="Delete_item" 
                                />  
                            </div>
                        </div>  
                    </td>                  
                </tr>
                
            );
        })
    );
    return(
        <div className="View">
            <h2>Shopping Lists:</h2>
            <div className="col-sm-12">
                <table className="table bordered">
                    <thead>
                        <tr>  
                            <td>LIST ID</td>
                            <td>LISTNAME</td>
                            <td>ACTIONS</td>
                        </tr>
                    </thead>
                    <tbody>
                        {renderShoppinglists()}
                    </tbody>
                </table>
                {getErrorMessage}
                {searchErrorMessage}
            </div>
        </div>
    );
}
export default ViewLists;
