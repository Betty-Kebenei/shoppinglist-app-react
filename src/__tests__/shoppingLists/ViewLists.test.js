import React from 'react';
import { shallow, mount, render } from 'enzyme';

import Config from '../../Config';

import ViewLists from '../../components/shoppinglists/ViewLists';

describe('interface of ViewLists component', () => {
    it('should render', () => {
        const wrapper = shallow(<ViewLists />);
        expect(wrapper).toMatchSnapshot();
    });
});