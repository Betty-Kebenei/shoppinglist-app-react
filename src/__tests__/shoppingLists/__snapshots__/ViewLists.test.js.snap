// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interface of ViewLists component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ViewLists />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Shopping Lists:
        </h2>,
        <div
          className="col-sm-12"
        >
          <table
            className="table bordered"
          >
            <thead>
              <tr>
                <td>
                  LIST ID
                </td>
                <td>
                  LISTNAME
                </td>
                <td>
                  ACTIONS
                </td>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>,
      ],
      "className": "View",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Shopping Lists:",
        },
        "ref": null,
        "rendered": "Shopping Lists:",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <table
              className="table bordered"
            >
              <thead>
                <tr>
                  <td>
                    LIST ID
                  </td>
                  <td>
                    LISTNAME
                  </td>
                  <td>
                    ACTIONS
                  </td>
                </tr>
              </thead>
              <tbody />
            </table>,
            undefined,
            undefined,
          ],
          "className": "col-sm-12",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <thead>
                  <tr>
                    <td>
                      LIST ID
                    </td>
                    <td>
                      LISTNAME
                    </td>
                    <td>
                      ACTIONS
                    </td>
                  </tr>
                </thead>,
                <tbody />,
              ],
              "className": "table bordered",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <tr>
                    <td>
                      LIST ID
                    </td>
                    <td>
                      LISTNAME
                    </td>
                    <td>
                      ACTIONS
                    </td>
                  </tr>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <td>
                        LIST ID
                      </td>,
                      <td>
                        LISTNAME
                      </td>,
                      <td>
                        ACTIONS
                      </td>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "LIST ID",
                      },
                      "ref": null,
                      "rendered": "LIST ID",
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "LISTNAME",
                      },
                      "ref": null,
                      "rendered": "LISTNAME",
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "ACTIONS",
                      },
                      "ref": null,
                      "rendered": "ACTIONS",
                      "type": "td",
                    },
                  ],
                  "type": "tr",
                },
                "type": "thead",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": "tbody",
              },
            ],
            "type": "table",
          },
          undefined,
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Shopping Lists:
          </h2>,
          <div
            className="col-sm-12"
          >
            <table
              className="table bordered"
            >
              <thead>
                <tr>
                  <td>
                    LIST ID
                  </td>
                  <td>
                    LISTNAME
                  </td>
                  <td>
                    ACTIONS
                  </td>
                </tr>
              </thead>
              <tbody />
            </table>
          </div>,
        ],
        "className": "View",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Shopping Lists:",
          },
          "ref": null,
          "rendered": "Shopping Lists:",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <table
                className="table bordered"
              >
                <thead>
                  <tr>
                    <td>
                      LIST ID
                    </td>
                    <td>
                      LISTNAME
                    </td>
                    <td>
                      ACTIONS
                    </td>
                  </tr>
                </thead>
                <tbody />
              </table>,
              undefined,
              undefined,
            ],
            "className": "col-sm-12",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <thead>
                    <tr>
                      <td>
                        LIST ID
                      </td>
                      <td>
                        LISTNAME
                      </td>
                      <td>
                        ACTIONS
                      </td>
                    </tr>
                  </thead>,
                  <tbody />,
                ],
                "className": "table bordered",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <tr>
                      <td>
                        LIST ID
                      </td>
                      <td>
                        LISTNAME
                      </td>
                      <td>
                        ACTIONS
                      </td>
                    </tr>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <td>
                          LIST ID
                        </td>,
                        <td>
                          LISTNAME
                        </td>,
                        <td>
                          ACTIONS
                        </td>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "LIST ID",
                        },
                        "ref": null,
                        "rendered": "LIST ID",
                        "type": "td",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "LISTNAME",
                        },
                        "ref": null,
                        "rendered": "LISTNAME",
                        "type": "td",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "ACTIONS",
                        },
                        "ref": null,
                        "rendered": "ACTIONS",
                        "type": "td",
                      },
                    ],
                    "type": "tr",
                  },
                  "type": "thead",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [],
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": "tbody",
                },
              ],
              "type": "table",
            },
            undefined,
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
